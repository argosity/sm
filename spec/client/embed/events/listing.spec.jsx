import { React, Snapshot, getScreenInstance } from 'hippo/testing/index';
import { delay } from 'lodash';
import MockDate from 'mockdate';

import EventModel from 'sm/models/embed/event';

import Listing from 'sm/embed/events/listing';
import { observable } from 'mobx';
import DATA from './data.json';


describe('Embedded Events Listing', () => {
    let events;
    let listing;
    beforeEach(() => {
        events = observable.array(DATA.map(ev => new EventModel(ev)));
        MockDate.set('2017-05-01');
        listing = mount(<Listing events={events} />);
        // grommit layer needs container
        document.body.appendChild(document.createElement('div'));
    });
    afterEach(() => {
        MockDate.reset();
    });

    it('renders events and matches snapshot', () => {
        DATA.forEach(ev =>
            expect(listing).toHaveRendered(`[data-event-identifier="${ev.identifier}"]`),
        );
        expect(Snapshot(<Listing events={events} />)).toMatchSnapshot();
    });

    it('displays info for events that have a page', () => {
        expect(listing).toHaveRendered('[data-event-identifier="cwMx65DgEAfG"] InfoButton Button');
        expect(listing).not.toHaveRendered('[data-event-identifier="sehR4uN224XV"] InfoButton Button');
    });

    it('displays purchase for events that are purchasable', () => {
        expect(listing).toHaveRendered('[data-event-identifier="cwMx65DgEAfG"] PurchaseButton Button');
        expect(listing).not.toHaveRendered('[data-event-identifier="sehR4uN224XV"] PurchaseButton Button');
    });

    it('can display info', () => {
        expect(listing).not.toHaveRendered('Information');
        listing.find('[data-event-identifier="cwMx65DgEAfG"] InfoButton Button').simulate('click');
        expect(listing).toHaveRendered('Information');
        const layer = document.querySelector('.grommetux-layer__container');
        expect(layer.textContent).toContain('test of stuff');
        layer.querySelector('.grommetux-layer__closer button').click();
        expect(document.querySelector('.grommetux-layer__container')).toBeNull();
    });

    it('can display purchase from info', () => {
        listing.find('[data-event-identifier="cwMx65DgEAfG"] InfoButton Button').simulate('click');
        document.querySelector('.grommetux-layer__container footer button').click();
    });

});
