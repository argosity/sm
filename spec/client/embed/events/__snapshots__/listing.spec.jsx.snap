// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embedded Events Listing renders events and matches snapshot 1`] = `
<div
  className="hippo events-listing"
>
  <h1>
    3
     events
  </h1>
  <div
    className="event"
    data-event-identifier="TnXMWyEL2dFe"
  >
    <img
      className="image"
      src="undefined/apiundefined/km2t/7t9x/xvdt/116057679f0ed7de0735437f45a635af.jpg"
    />
    <div
      className="info"
    >
      <div
        className="title"
      >
        <h2>
          Radiohead
        </h2>
      </div>
      <h3
        className="sub-title"
      >
        with Giant Domino
      </h3>
      <p
        className="description"
      >
        Innovative motivating circuit
      </p>
      <div
        className="event-occurrences"
      >
        <ul
          className="dates"
        >
          <li
            className="date"
          >
            1:56pm Aug 24th 2017
          </li>
          <li
            className="date"
          >
            4:59am Aug 25th 2017
          </li>
          <li
            className="date"
          >
            9:59am Aug 26th 2017
          </li>
        </ul>
      </div>
    </div>
    <div
      className="actions"
    />
  </div>
  <div
    className="event"
    data-event-identifier="sehR4uN224XV"
  >
    <img
      className="image"
      src="undefined/apiundefined/cf32/3571/qez4/e7aa365117af066cf62b115394f5eb4b.png"
    />
    <div
      className="info"
    >
      <div
        className="title"
      >
        <h2>
          Little Feat
        </h2>
      </div>
      <h3
        className="sub-title"
      >
        with Dark Venom
      </h3>
      <p
        className="description"
      >
        Automated needs-based open system
      </p>
      <div
        className="event-occurrences"
      >
        <ul
          className="dates"
        >
          <li
            className="date"
          >
            9:59am Aug 26th 2017
          </li>
          <li
            className="date"
          >
            4:10am Sep 2nd 2017
          </li>
        </ul>
      </div>
    </div>
    <div
      className="actions"
    />
  </div>
  <div
    className="event"
    data-event-identifier="cwMx65DgEAfG"
  >
    <img
      className="image"
      src="undefined/apiundefined/bjp5/c4sd/gmud/3fad2e44aac60c2a0bf56f4418c33235.png"
    />
    <div
      className="info"
    >
      <div
        className="title"
      >
        <h2>
          Genesis
        </h2>
      </div>
      <h3
        className="sub-title"
      >
        with Molten Man II
      </h3>
      <p
        className="description"
      >
        Right-sized bottom-line customer loyalty
      </p>
      <div
        className="event-occurrences"
      >
        <ul
          className="dates"
        >
          <li
            className="date"
          >
            9:59am Aug 26th 2017
          </li>
          <li
            className="date"
          >
            4:10am Sep 2nd 2017
          </li>
        </ul>
      </div>
    </div>
    <div
      className="actions"
    >
      <button
        aria-label={undefined}
        className="grommetux-button"
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        target="_blank"
        type="button"
      >
        <span
          className="grommetux-button__icon"
        >
          <svg
            aria-label="circle-information"
            className="grommetux-control-icon grommetux-control-icon-circle-information grommetux-control-icon--responsive"
            height="24px"
            role="img"
            version="1.1"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,10 L12,18 M12,6 L12,8"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </span>
        <span
          className="grommetux-button__label"
        >
          Information
        </span>
      </button>
      <button
        aria-label={undefined}
        className="grommetux-button"
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <span
          className="grommetux-button__icon"
        >
          <svg
            aria-label="credit-card"
            className="grommetux-control-icon grommetux-control-icon-credit-card grommetux-control-icon--responsive"
            height="24px"
            role="img"
            version="1.1"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M1,5.00087166 C1,4.4481055 1.43945834,4 2.00246167,4 L21.9975383,4 C22.5511826,4 23,4.44463086 23,5.00087166 L23,18.9991283 C23,19.5518945 22.5605417,20 21.9975383,20 L2.00246167,20 C1.44881738,20 1,19.5553691 1,18.9991283 L1,5.00087166 Z M1,8 L23,8 L23,10 L1,10 L1,8 Z M5,15 L7,15 L7,15.5 L5,15.5 L5,15 Z M10,15 L16,15 L16,15.5 L10,15.5 L10,15 Z"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </span>
        <span
          className="grommetux-button__label"
        >
          Purchase
        </span>
      </button>
    </div>
  </div>
</div>
`;
